File: androidApp/src/main/java/com/cacd2/cacdgame/android/features/history/ui/GameHistoryEmptyView.kt
package com.cacd2.cacdgame.android.features.history.ui

import androidx.compose.foundation.Image
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material.Surface
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.cacd2.cacdgame.android.AppTheme
import com.cacd2.cacdgame.android.R
import com.cacd2.cacdgame.android.fonts

@Composable
fun GameHistoryEmptyView(modifier: Modifier = Modifier) {
    Column(
        modifier = modifier,
        verticalArrangement = Arrangement.Center,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Image(painter = painterResource(id = R.drawable.history_empty),
            contentDescription = "")
        Text(
            text = stringResource(id = R.string.history_emptylist),
            modifier = Modifier.padding(20.dp),
            textAlign = TextAlign.Center,
            style =
            TextStyle(
                fontFamily = fonts,
                fontSize = 18.sp,
                fontWeight = FontWeight.SemiBold
            )
        )
    }
}

@Preview
@Composable
fun GameHistoryEmptyPrevView() {
    AppTheme {
        Surface {
            GameHistoryEmptyView(modifier = Modifier.fillMaxSize())
        }
    }
}

File: iosApp/iosApp/Features/History/UI/GameHistoryEmptyView.swift
//
//  GameHistoryEmptyView.swift
//  CACDGAME
//
//  Created by François Dabonot on 06/12/2023.
//  Copyright © 2023 CACD2. All rights reserved.
//

import SwiftUI

struct GameHistoryEmptyView: View {
    var body: some View {
        VStack(alignment: .center, content: {
            R.image.history_empty.image
            Text(R.string.localizable.history_emptylist.localizedKey)
                .multilineTextAlignment(.center)
                .padding()
                .font(R.font.poppinsSemiBold.font(size: 18))

        })
    }
}

#Preview {
    GameHistoryEmptyView()
}
